internal/v2/application/command.go

```diff
@@ -617,22 +634,19 @@ func createCommandValueFromDeviceResource(dr models.DeviceResource, v string) (*
 			} else if math.IsNaN(val) {
 				err = fmt.Errorf("fail to parse %v to float64, unexpected result %v", v, val)
 			} else {
-				result, err = dsModels.NewFloat64Value(dr.Name, origin, val)
+				result, err = dsModels.NewCommandValue(dr.Name, v2.ValueTypeFloat64, val)
 			}
 		}
-	case strings.ToLower(v2.ValueTypeFloat64Array):
+	case v2.ValueTypeFloat64Array:
 		var arr []float64
-		err = json.Unmarshal([]byte(v), &arr)
+		err := json.Unmarshal([]byte(v), &arr)
 		if err != nil {
-			return result, err
+			errMsg := fmt.Sprintf("failed to convert set parameter %s to ValueType %s", v, dr.Properties.ValueType)
+			return result, edgexErr.NewCommonEdgeX(edgexErr.KindServerError, errMsg, err)
 		}
-		result, err = dsModels.NewFloat64ArrayValue(dr.Name, origin, arr)
+		result, err = dsModels.NewCommandValue(dr.Name, v2.ValueTypeFloat64Array, arr)
 	default:
-		err = errors.New("unsupported deviceResource value type")
-	}
-
-	if err != nil {
-		return result, err
+		err = edgexErr.NewCommonEdgeX(edgexErr.KindServerError, "unrecognized value type", nil)
```

Comment:
wrap the original err into the new one

Reply:
err = edgexErr.NewCommonEdgeX(edgexErr.KindServerError, "unrecognized value type", err)

Reply:
there will be no original error to wrap if switch case goes to default case